include/nuttx/efuse/efuse.h

```diff
@@ -0,0 +1,294 @@
+ *              of bits to be read from efuses.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_READ_FIELD_BLOB      _EFUSEIOC(0x0001)
+
+/* Command:     EFUSEIOC_READ_FIELD_BIT
+ * Description: Read of state of an efuse bit.
+ * Arguments:   A structure containing the field to be read.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_READ_FIELD_BIT       _EFUSEIOC(0x0002)
+
+/* Command:     EFUSEIOC_WRITE_FIELD_BLOB
+ * Description: Write a blob of bits to an efuse's field
+ * Arguments:   A structure containing the field[], the src memory and the
+ *              amount of bits to write.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_WRITE_FIELD_BLOB     _EFUSEIOC(0x0003)
+
+/* Command:     EFUSEIOC_WRITE_FIELD_BIT
+ * Description: Write a bit to an efuse (burn it).
+ * Arguments:   A structure containing bit field.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_WRITE_FIELD_BIT      _EFUSEIOC(0x0004)
+
+/* Command:     EFUSEIOC_GET_FIELD_SIZE
+ * Description: Get the length of the field in bits.
+ * Arguments:   A structure containing the fields.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_GET_FIELD_SIZE       _EFUSEIOC(0x0005)
+
+/* Command:     EFUSEIOC_READ_REG
+ * Description: Read an efuse register.
+ * Arguments:   A structure containing the block number and the register to
+ *              be read.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_READ_REG             _EFUSEIOC(0x0006)
+
+/* Command:     EFUSEIOC_WRITE_REG
+ * Description: Write an efuse register.
+ * Arguments:   A structure containing the block number, the register to
+ *              write and value to be written.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_WRITE_REG            _EFUSEIOC(0x0007)
+
+/* Command:     EFUSEIOC_READ_BLOCK
+ * Description: Read a key from efuse's block.
+ * Arguments:   A structure containing the block number, the dst_key pointer,
+ *              the offset in bits and the amount of bits to read.
+ * Return:      Zero (OK) on success.  Minus one will be returned on failure
+ *              with the errno value set appropriately.
+ */
+
+#define EFUSEIOC_READ_BLOCK           _EFUSEIOC(0x0008)
```

Comment:
@acassis what's difference between field/bit and block/reg? can we unify the read/write interface into one?

Reply:
Hi @xiaoxiang781216 read/write a field bit will work with fields of a single bit (a single fuse itself), the operation over reg will take care of 32 fuses just like a ordinary 32-bit register and finally block is an arbitrary size of fuse. This driver was based on existing Espressif eFuse driver, but I agree with you, we could come up with a single operation that will ignore these differences.

Reply:
it's better to define a simple and clean interface first, then consider how to map the generic interface to espressif driver api. Otherwise,  other vendor is hard to understand the interface and then provide a reasonable implementation.